<launch>
  <!-- Setze Pfad zu CAD-Modellen -->
  <env name="GAZEBO_MODEL_PATH" value="$(find my_objects)"/>

  <!-- Starte Gazebo mit der eigenen Welt -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find multi_robot_station)/worlds/my_world.world" />
  </include>

  <!-- Robot 1 -->
  <group ns="sero_1">
    <!-- Load robot_description globally -->
    <param name="/sero_1/robot_description" command="cat $(find sero_1)/urdf/sero_1.urdf" />
    
    <!-- This model reads from the parameter set above -->
    <node name="spawn_sero_1" pkg="gazebo_ros" type="spawn_model"
          args="-param /sero_1/robot_description -urdf -model sero_1 -x 1.5 -y 0.0 -z 0.525 -R 0 -P 0 -Y 1.5708"
          output="screen"/>
    
    <!-- State publisher reads the same URDF -->
    <node name="rsp_sero_1" pkg="robot_state_publisher" type="robot_state_publisher" output="screen">
      <param name="robot_description" command="cat $(find sero_1)/urdf/sero_1.urdf" />
    </node>
  </group>


  <!-- Robot 2 -->
  <group ns="sero_2">
    <!-- Load robot_description globally -->
    <param name="/sero_2/robot_description" command="cat $(find sero_2)/urdf/sero_2.urdf" />
    
    <!-- This model reads from the parameter set above -->
    <node name="spawn_sero_2" pkg="gazebo_ros" type="spawn_model"
          args="-param /sero_2/robot_description -urdf -model sero_2 -x 0.0 -y -1.5 -z 0.525 -R 0 -P 0 -Y 0"
          output="screen"/>
    
    <!-- State publisher reads the same URDF -->
    <node name="rsp_sero_2" pkg="robot_state_publisher" type="robot_state_publisher" output="screen">
      <param name="robot_description" command="cat $(find sero_2)/urdf/sero_2.urdf" />
    </node>
  </group>


  <!-- Robot 3 -->
  <group ns="sero_3">
    <!-- Load robot_description globally -->
    <param name="/sero_3/robot_description" command="cat $(find sero_3)/urdf/sero_3.urdf" />
    
    <!-- This model reads from the parameter set above -->
    <node name="spawn_sero_3" pkg="gazebo_ros" type="spawn_model"
          args="-param /sero_3/robot_description -urdf -model sero_3 -x 0.0 -y 0.0 -z 0.525 -R 0 -P 0 -Y 0"
          output="screen"/>
    
    <!-- State publisher reads the same URDF -->
    <node name="rsp_sero_3" pkg="robot_state_publisher" type="robot_state_publisher" output="screen">
      <param name="robot_description" command="cat $(find sero_3)/urdf/sero_3.urdf" />
    </node>
  </group>


  <node name="spawn_workobject" pkg="gazebo_ros" type="spawn_model"
      args="-file $(find workobject)/urdf/workobject.urdf -urdf -model workobject -x 0.0 -y 1.0 -z 0.531 -R 3.1415 -P 0 -Y 0.0"
      output="screen" />

  <!-- Attacher-Service -->
  <node pkg="gazebo_ros_link_attacher" type="gazebo_ros_link_attacher_node" name="link_attacher_node" output="screen" />

  <!-- Python-Skript zum Verbinden -->
  <node pkg="multi_robot_station" type="attach_robots.py" name="attach_robots" output="screen" />

</launch>
